engineio/src/asynchronous/async_socket.rs

```diff
@@ -202,6 +204,23 @@ impl Socket {
         *self.last_ping.lock().await = Instant::now();
     }
 
+    // Returns the time in milliseconds that is left until a new ping must be received.
+    // This is used to detect whether we have been disconnected from the server.
+    // See https://socket.io/docs/v4/how-it-works/#disconnection-detection
+    async fn time_to_next_ping(&self) -> u64 {
+        match Instant::now().checked_duration_since(*self.last_ping.lock().await) {
+            Some(since_last_ping) => {
+                let since_last_ping = (since_last_ping.as_millis() >> 64) as u64;
```

Comment:
Am I mistaken here but... is right shifting by 64 almost always going to be 0?

Reply:
Sus.